InAppUtils/InAppUtils.m

```diff
@@ -26,50 +25,92 @@ - (dispatch_queue_t)methodQueue
 
 RCT_EXPORT_MODULE()
 
+NSString *const IAPProductsEvent = @"IAP-products";
+NSString *const IAPTransactionEvent = @"IAP-transaction";
+NSString *const IAPRestoreEvent = @"IAP-restore";
+
+- (NSDictionary *)constantsToExport
+{
+  return @{ 
+      @"IAPProductsEvent": IAPProductsEvent, 
+      @"IAPTransactionEvent": IAPTransactionEvent, 
+      @"IAPRestoreEvent": IAPRestoreEvent 
+  };
+}
+
+- (NSArray<NSString *> *)supportedEvents
+{
+  return @[IAPProductsEvent, IAPTransactionEvent, IAPTransactionEvent];
+}
+
+/**
+    GUIDE TO LISTENERS:
+    source: https://facebook.github.io/react-native/docs/native-modules-ios.html
+
+    import { NativeEventEmitter, NativeModules } from 'react-native';
+    const { InAppUtils } = NativeModules;
+
+    const IAPEmitter = new NativeEventEmitter(InAppUtils);
+
+    const subscription = IAPEmitter.addListener(IAPEmitter.IAPProductsEvent,
+        (response) => console.log(response.state, response) // response can contain error or products dependent on state
+    );
+    ...
+    // Don't forget to unsubscribe, typically in componentWillUnmount
+    subscription.remove();
+*/
+
+// Will be called when this module's first listener is added.
+-(void)startObserving {
+    hasListeners = YES;
+    // Set up any upstream listeners or background tasks as necessary
+}
+
+// Will be called when this module's last listener is removed, or on dealloc.
+-(void)stopObserving {
+    hasListeners = NO;
+    // Remove upstream listeners, stop unnecessary background tasks
+}
+
 - (void)paymentQueue:(SKPaymentQueue *)queue
  updatedTransactions:(NSArray *)transactions
 {
     for (SKPaymentTransaction *transaction in transactions) {
         switch (transaction.transactionState) {
-            case SKPaymentTransactionStateFailed: {
-                NSString *key = RCTKeyForInstance(transaction.payment.productIdentifier);
-                RCTResponseSenderBlock callback = _callbacks[key];
-                if (callback) {
+            case SKPaymentTransactionStateFailed: {   
```

Comment:
Is het mogelijk om zo minmogelijk code in obj c te doen, en gewoon een updatedTransactions listener in RN? Dan kunnen we vanuit daar namelijk meteen loggen, etc.

Reply:
Dit zou dus betekenen; de hele switch() verwijderen